go/common/identity/identity.go

```diff
@@ -28,33 +30,132 @@ const (
 
 	tlsKeyFilename  = "tls_identity.pem"
 	tlsCertFilename = "tls_identity_cert.pem"
+
+	// These are used for the sentry client connection to the sentry node and are never rotated.
+	tlsSentryClientKeyFilename  = "sentry_client_tls_identity.pem"
+	tlsSentryClientCertFilename = "sentry_client_tls_identity_cert.pem"
 )
 
+// ErrCertificateRotationForbidden is returned by RotateCertificates if
+// TLS certificate rotation is forbidden.  This happens when rotation is
+// enabled and an existing TLS certificate was successfully loaded
+// (or a new one was generated and persisted to disk).
+var ErrCertificateRotationForbidden = errors.New("identity", 1, "identity: TLS certificate rotation forbidden")
+
 // Identity is a node identity.
 type Identity struct {
+	sync.RWMutex
+
 	// NodeSigner is a node identity key signer.
 	NodeSigner signature.Signer
 	// P2PSigner is a node P2P link key signer.
 	P2PSigner signature.Signer
 	// ConsensusSigner is a node consensus key signer.
 	ConsensusSigner signature.Signer
-	// TLSSigner is a node TLS certificate signer.
-	TLSSigner signature.Signer
-	// TLSCertificate is a certificate that can be used for TLS.
-	TLSCertificate *tls.Certificate
+	// tlsSigner is a node TLS certificate signer.
+	tlsSigner signature.Signer
```

Comment:
As a matter of style I prefer private struct members to be separated from public ones (and not interleaved like it is here).